legion/engine/core/scenemanagement/scenemanager.cpp

```diff
@@ -1,69 +1,178 @@
 
+    bool SceneManager::createScene(const std::string& name)
+    {
+        auto sceneEntity = m_ecs->createEntity();
+        std::vector<ecs::entity_handle> children;
+        for (size_type i = 0; i < m_ecs->world.child_count(); i++)
+        {
+            children.push_back(m_ecs->world.get_child(i));
+        }
+        for (auto child : children)
+        {
+            if (child == sceneEntity)
+                continue;
+            child.set_parent(sceneEntity);
+        }
 
+        if (!getScene(name))
+        {
+            scene s;
+            s.id = nameHash(name);
+            sceneNames.emplace(s.id, name);
+            sceneEntity.add_component<scene>(s);
+            sceneList.emplace(nameHash(name), sceneEntity);
+        }
+        return SceneManager::saveScene(name, sceneEntity);
+    }
 
-    bool SceneManager::createScene(const std::string& name)
+    bool SceneManager::importResource(const std::string& path, bool unique, const std::string& unique_ident)
+    {
+        std::string key = path;
+        if(unique)
+            key += unique_ident;
+
+        auto extension = path.substr(path.rfind('.'),std::string::npos);
+
+        //ask the AssetImporter to prefetch the data
+
+        //TODO(algo-ryth-mix): prefetching is prolly still a good idea, but for now we want this working
+        #if 0
+        return filesystem::AssetImporter::prefetch(fs::view(path),key);
+        #endif
+        return true;
+    }
+
+
+
+    bool SceneManager::prefetchRecursive(nlohmann::json j, id_type hash)
     {
-        scene s;
-        s.id = nameHash(name);
-        sceneNames.emplace(s.id, name);
-        auto sceneHandle = m_ecs->createEntity().add_component<scenemanagement::scene>(s);
-        sceneList.emplace(nameHash(name), sceneHandle);
+
+        // prefetch for all children
+        for (auto& [key, value] : j.items())
+        {
+            if(key.empty() || !(value.is_object() || value.is_array())) continue;
+            id_type this_hash = hash_combine(nameHash(key), hash);
+            prefetchRecursive(value, this_hash);
+        }
+
+        // check if current node has a Filepath attribute
+        const auto itr = j.find("Filepath");
+        if (itr != j.end())
+        {
+
+            //check if current node wants a unique resource
+            bool unique = false;
+            const auto unique_itr = j.find("unique");
+
+            if (unique_itr != j.end())
+            {
+                unique = unique_itr->get<bool>();
+            }
+            const std::string path = itr->get<std::string>();
+
+            log::debug("loading path: {}",path);
+            //import 
+            if(!importResource(path, unique, std::to_string(hash)))
+            {
+                log::error("failed to load {}",path);
+                return false;
+            }
+            log::info("successfully loaded {}",path);
+        }
         return true;
+
+    }
```

Comment:
Pretty sure we don't use this any more

Reply:
it is not, we should think of a different prefetching method however, since loading stuff in place can be quite costly. maybe a `earlyLoad` method where the resources are just fetched but not loaded ?